apps/ui/src/components/App/Nav.vue

```diff
@@ -26,8 +26,11 @@ type NavigationItem = {
   active?: boolean;
 };
 
+const emit = defineEmits<{
+  (e: 'navigated');
+}>();
+
```

Comment:
I feel like it's ugly to have to propagate this event up from those components to handle this.

It would be much cleaner to just watch route changes and I think it should give us the same result.

Reply:
This is to decouple the sidebar open/close handling from those 3 components (Nav, Sidebar and Topnav). (4 components after #798)

Logically, these component should not be aware of the existence of the sidebar, which should be handle at the layout level.

One more obscurev way to avoid repetition, and keep things logic, is to handle that inside AppLink, but will be less verbose and obvious: we add an `click` event inside the `router-link` in AppLink, to always close the sidebar

Reply:
We are not really decoupling it now, because those components now need to expose `navigated` event.

We can make those components truly independent and locate logic of what happens after navigation by using [`afterEach`](https://router.vuejs.org/guide/advanced/navigation-guards.html#Global-After-Hooks) in `LayoutApp`.

Reply:
Very good idea, just did that